File: PRACTICA#3/PERSISTENCIA/1/JAVA/ArchivoEmpleado.java
import java.io.*;

public class ArchivoEmpleado {
    private String nomA;

    public ArchivoEmpleado(String nomA) {
        this.nomA = nomA;
    }

    public void crearArchivo() {
        try (FileWriter f = new FileWriter(nomA)) {
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public void guardarEmpleado(Empleado e) {
        try (FileWriter f = new FileWriter(nomA, true)) {
            f.write(e.getNombre() + "," + e.getEdad() + "," + e.getSalario() + "\n");
        } catch (IOException ex) {
            ex.printStackTrace();
        }
    }

    public Empleado buscaEmpleado(String n) {
        try (BufferedReader f = new BufferedReader(new FileReader(nomA))) {
            String line;
            while ((line = f.readLine()) != null) {
                String[] parts = line.split(",");
                if (parts[0].equals(n)) {
                    return new Empleado(parts[0], Integer.parseInt(parts[1]), Float.parseFloat(parts[2]));
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }

    public Empleado mayorSalario(float s) {
        try (BufferedReader f = new BufferedReader(new FileReader(nomA))) {
            String line;
            while ((line = f.readLine()) != null) {
                String[] parts = line.split(",");
                if (Float.parseFloat(parts[2]) > s) {
                    return new Empleado(parts[0], Integer.parseInt(parts[1]), Float.parseFloat(parts[2]));
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: PRACTICA#3/PERSISTENCIA/1/PYTHON/ArchivoEmpleado.py
from Empleado import Empleado

class ArchivoEmpleado:
    def __init__(self, nomA: str):
        self.nomA = nomA
    
    def crearArchivo(self):
        with open(self.nomA, 'w') as f:
            f.write("")
    
    def guardarEmpleado(self, e: 'Empleado'):
        with open(self.nomA, 'a') as f:
            f.write(f"{e.nombre},{e.edad},{e.salario}\n")
    
    def buscaEmpleado(self, n: str):
        with open(self.nomA, 'r') as f:
            for line in f:
                nombre, edad, salario = line.strip().split(',')
                if nombre == n:
                    return Empleado(nombre, int(edad), float(salario))
        return None
    
    def mayorSalario(self, s: float):
        with open(self.nomA, 'r') as f:
            for line in f:
                nombre, edad, salario = line.strip().split(',')
                if float(salario) > s:
                    return Empleado(nombre, int(edad), float(salario))
        return None